alternative-1-iterate-generic.Rmd

```diff
@@ -13,58 +13,62 @@ knitr::opts_chunk$set(
 
 # R Iterator Proposal: `iterate()` generic.
 
-R doesn't currently have a mechanism for users to customize what happens
-when objects are passed to `for`. Below is a proposal for what such a
-mechanism might look like:
-
--   R provides a generic with signature: `iterate(x, state)`.
-
--   User implements a `iterate()` method for their object, which is
-    called with arguments `iterate(x, state)`, and expected to return a
-    list of length 2: `list(next_elem, next_state)`.
-
--   `for` repeatedly calls `iterate(x, state)` until `NULL` is returned.
-
-(This implementation is very similar to the Julia approach).
-
-The equivalent of the following could be added to base R (with core
-parts implemented in C):
+Inspired by
```

Comment:
I restructured the presentation of the alternatives to focus on the new/different bits